packages/discord.js/src/errors/ErrorCodes.js

```diff
@@ -0,0 +1,296 @@
+
+ * @property {string} InteractionCollectorError
+
+ * @property {string} FileNotFound
+
+ * @property {string} UserBannerNotFetched
+ * @property {string} UserNoDMChannel
+
+ * @property {string} VoiceNotStageChannel
+
+ * @property {string} VoiceStateNotOwn
+ * @property {string} VoiceStateInvalidType
+
+ * @property {string} ReqResourceType
+
+ * @property {string} ImageFormat
+ * @property {string} ImageSize
+
+ * @property {string} MessageBulkDeleteType
+ * @property {string} MessageNonceType
+ * @property {string} MessageContentType
+
+ * @property {string} SplitMaxLen
+
+ * @property {string} BanResolveId
+ * @property {string} FetchBanResolveId
+
+ * @property {string} PruneDaysType
+
+ * @property {string} GuildChannelResolve
+ * @property {string} GuildVoiceChannelResolve
+ * @property {string} GuildChannelOrphan
+ * @property {string} GuildChannelUnowned
+ * @property {string} GuildOwned
+ * @property {string} GuildMembersTimeout
+ * @property {string} GuildUncachedMe
+ * @property {string} ChannelNotCached
+ * @property {string} StageChannelResolve
+ * @property {string} GuildScheduledEventResolve
+ * @property {string} FetchOwnerId
+
+ * @property {string} InvalidType
+ * @property {string} InvalidElement
+
+ * @property {string} MessageThreadParent
+ * @property {string} MessageExistingThread
+ * @property {string} ThreadInvitableType
+
+ * @property {string} WebhookMessage
+ * @property {string} WebhookTokenUnavailable
+ * @property {string} WebhookURLInvalid
+ * @property {string} WebhookApplication
+ * @property {string} MessageReferenceMissing
+
+ * @property {string} EmojiType
+ * @property {string} EmojiManaged
+ * @property {string} MissingManageEmojisAndStickersPermission
+ * @property {string} NotGuildSticker
+
+ * @property {string} ReactionResolveUser
+
+ * @property {string} VanityURL
+
+ * @property {string} InviteResolveCode
+
+ * @property {string} InviteNotFound
+
+ * @property {string} DeleteGroupDMChannel
+ * @property {string} FetchGroupDMChannel
+
+ * @property {string} MemberFetchNonceLength
+
+ * @property {string} GlobalCommandPermissions
+ * @property {string} GuildUncachedEntityResolve
+
+ * @property {string} InteractionAlreadyReplied
+ * @property {string} InteractionNotReplied
+ * @property {string} InteractionEphemeralReplied
+
+ * @property {string} CommandInteractionOptionNotFound
+ * @property {string} CommandInteractionOptionType
+ * @property {string} CommandInteractionOptionNoSubcommand
+ * @property {string} CommandInteractionOptionNoSubcommandGroup
+ * @property {string} AutocompleteInteractionOptionNoFocusedOption
+
+ * @property {string} ModalSubmitInteractionFieldNotFound
+ * @property {string} ModalSubmitInteractionFieldType
+
+ * @property {string} InvalidMissingScopes
+
+ * @property {string} NotImplemented
+
+ * @property {string} SweepFilterReturn
+ */
+
+// JSDoc for intellisense purposes
+/**
+ * @type {DiscordjsErrorCodes}
+ * @ignore
+ */
+module.exports = createEnum([
```

Comment:
Will we be adding new messages in the middle of the enum? If so, it will be an issue for people who will use magic numbers. I suggest separate the groups by hundreds or thousands if we will be adding new message in the middle.
```
0 - ClientInvalidOption
3 - ClientNotReady
1000 - TokenInvalid
1002 - ApplicationCommandPermissionsTokenMissing
2000 - WSCloseRequested
2003 - ManagerDestroyed
```

Reply:
This is a bit of a tricky question. I think realistically we aren't able to add elements in the middle for that reason..? (i.e. it would breaking)

In the event that we can and it's not considered breaking, we probably just don't care, frankly. People shouldn't be using magic numbers in the first place and if they're using TS their code won't even compile.